src/main/java/org/opentripplanner/routing/edgetype/StreetEdgeTraversal.java

```diff
@@ -0,0 +1,282 @@
+package org.opentripplanner.routing.edgetype;
+
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.StateEditor;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.util.ElevationUtils;
+import org.opentripplanner.routing.vertextype.IntersectionVertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by kathrynkillebrew on 1/23/15.
+ */
+public class StreetEdgeTraversal {
+
+    private static Logger LOG = LoggerFactory.getLogger(StreetEdgeTraversal.class);
+
+    private static final double GREENWAY_SAFETY_FACTOR = 0.1;
+
+    /** return a StateEditor rather than a State so that we can make parking/mode switch modifications for kiss-and-ride. */
+    public static StateEditor doTraverse(StreetEdge edge, State s0, RoutingRequest options, TraverseMode traverseMode) {
```

Comment:
I was considering that.  The function generally only gets called once from the `StreetEdge` traversal, with the exception of kiss-and-ride trips, and itself returns the state editor, which is the mutable thing used for tracking state, so I thought it might make sense to simply leave this part as a static method, rather than go create a new object to track.